packages/frameworks/src/frameworks.ts

```diff
@@ -1446,6 +1473,7 @@ export const frameworks = [
         dest: '/200.html',
       },
     ],
+    versionDependencies: ['nuxt3', 'nuxt3-edge'],
```

Comment:
Can we reuse the the existing `detectors` logic to get the matching version instead of this new property?

Reply:
I did look at using `detectors`, but as written currently they do not identify an actual package we can use to get the version. We could replace them all to use packages only, but I'm concerned that we'd break detection with such a big change. There may be specific frameworks where we really need to rely on that file detection--not sure. We could changes here incrementally, though.

Or, were you thinking of some other approach?

Reply:
Can you give an example of one that wouldn't work with `detectors`? I'm not sure I understand the problem.

Reply:
The current `blitzjs` detector looks for `path: 'blitz.config.(js|ts)'`. That will identify a given project as a Blitz.js project, but it doesn't give us any hint as to the version number. We'd still need to know that the package is `blitz` in order to lookup the installed version number.

We can either:

1. have a separate property that points to packages to find version numbers: this is what the PR currently does
2. update the detectors to look only for packages, then we know where to find the version numbers: not sure if this is possible for all `npm` packages, but definitely would have to have some flexibility for non-`npm` packages like `Hugo`
3. do something else?

I'd be happy to talk this through.

Reply:
I prefer option 3: Add a `frameworkVersion` prop to the `frameworks.ts` which would override the version found in package.json

That way you only need to add it to blitz, sveltkit, or others with two entries.

Reply:
@styfle Hrm. I think we're misunderstanding each other. I want the actual, specific version that the project happens to use. We can't know that until we detect it per deployment.

Reply:
@EndangeredMassa Ok in that case, I think its fine to ignore the frameworks that don't have a `package.json` listed in `detectors`. They don't need to list the version.

Reply:
To clarify, your suggestion is:

- leverage existing detectors, no new properties
- if a detector hits package.json
    - have it return the version range as well
    - if the version range is a specific version
        - use it
    - else
        - extract the matched package from the detector
        - lookup the installed value for the matched package
        - use it

Is that right?

It looks like 31/41 frameworks reference package.json dependencies. This would be reasonably good coverage.

NOTE: Remix, Sanity, and Hydrogen would not get version support given this change.

Reply:
Yes that algorithm sounds right. If we can do framework detection and version detection in the same pass, that would be ideal.

31/41 is a bit lower than expected but still reasonable. I think it is showing a deficiency in our detectors. For example, does remix always have a `remix.config.js`? Perhaps that should also have a dependency check.

Reply:
Updated in:
- [`6774f99` (#9009)](https://github.com/vercel/vercel/pull/9009/commits/6774f990b472b1e7c4298427084c6025deccd424)
- [`6ca6b39` (#9009)](https://github.com/vercel/vercel/pull/9009/commits/6ca6b397c9e9bd230fb75f46678ef87ba6866486)

I'm going to open a different PR to see if we can update some of the detectors for the frameworks that do not currently reference package names.

Reply:
New PR to update some of the detectors to use package names: https://github.com/vercel/vercel/pull/9167